### ISSUE TITLE
Feed from data source not working

### ISSUE THREAD

**Issue by A46webchair** [code4recovery/12-step-meeting-list#170]
Received new URL from District 2 (Santa Fe). The feed does not work. Neither the webmaster fro District 2 nor I know what to do next. This feed goes into https://nm-aa.org website.

https://santafeaa.org/wp-admin/admin-ajax.php?action=meetings&key=53e1d342a225a98f7ddaa2f52d7d18dd

This link worked at one point but is no longer working.  I have deleted and re-added the feed and that did not work either. Is the issue on the Santa Fe website side?

Any help would be appreciated. We are looking for some direction. Thank you

Kerri K, Area 46 Webchair
webchair@nm-aa.org

![image](https://user-images.githubusercontent.com/72042564/94473674-ba23fc80-0189-11eb-90fc-747940c0b5af.png)

**Comment by tech2serve** [comment]
@A46webchair,

I just used the Santa Fe link you provided and took in 138 meetings on my site. 

What version of TSML are you each using? Versions of Wordpress? Any errors showing up in the console?

**Comment by A46webchair** [comment]
TSML = Version 3.8.0
WordPress = Version 5.5.1

I have not seen any errors.

**Comment by A46webchair** [comment]
@tech2serve 

TSML = Version 3.8.0
WordPress = Version 5.5.1

I have not seen any errors.

**Comment by tech2serve** [comment]
Thanks. I need to keep asking some questions because I am not able to duplicate the issue. When this occurs, we often find the issue is ultimately with the browser or website/hosting platform. Please clear your browser cache, and the cache for your nm-aa.org website.

What browser are you using?

When you connect to the feed does it run to 100% and then not show meetings? Or does it not show a progress at all?

**Comment by A46webchair** [comment]
No problem.  Our host is Site 5 (their customer service is horrible). 

**Browser** - I am currently using Firefox.  Just tried in Chrome and still no meetings appear.  Cleared cache in Firefox. Still got no result.

I do not see any progress status bar anywhere.  The screen shot below shows where I am entering the link. This is the correct spot, right? Where should I see a status bar?

**FYI** - We connect to another feed as well (for Albuquerque) but I am currently NOT refreshing that feed.  I wanted to stay consistent in how I showed online meetings during the pandemic. So, unfortunately I cannot check to see if this feed is working as I am not ready to refresh it yet.


![image](https://user-images.githubusercontent.com/72042564/94492116-cbc7cd00-01a6-11eb-8dc4-ff6c02fc997a.png)

**Comment by tech2serve** [comment]
Just experienced another support request for which flushing/clearing the website cache resolved a unexplained issue. 

Were you able to contact Site 5 about clearing the website cache? You might also look into plugins that will clear the website cache, but I don't have any recommendations or experience with those.

**Comment by crispyatl** [comment]
For the record, I tried this myself on version 3.8.0 and it's working fine.

![image](https://user-images.githubusercontent.com/64294568/95814983-bc3d9d80-0ce9-11eb-9a11-0c88b84f1907.png)

**Comment by A46webchair** [comment]
Hey Everyone.  Just migrated to a new webhost (thought that would clear my website cache).

I have two meeting feeds and neither are working.  I am at a loss.  It is critical that I get these feeds working again.

Now on PHP 7.3.23
WP Version 5.5.1

**Comment by tech2serve** [comment]
Hello @A46webchair 

I just connected again to Santa Fe using the link you provided above. Everything seemed to work just fine. And given that @crispyatl was also able to import Santa Fe's JSON feed, I think we need to conclude something is amiss between you and Santa Fe. 

I suggest looking more closely at plugins being used on your site and the two sites that you're trying to connect with. It seems that Crispy may be correct about you getting blocked by them, which would have to be something specific to your IP or probably domain name. Maybe they are running a blacklisting checker? And "nm-aa.org" is blacklisted? Not really sure.

Also, please make sure to have both of those site and you check logs for any evidence of blocks or errors. Try opening up a developer console so you can see any errors that might be thrown in there when you start the connection process. I know you did that earlier, but clearly something isn't working. And ask the feed sites to do that same.

Can you enable WordPress debugging? This is the [normal method](https://wordpress.org/support/article/debugging-in-wordpress/) for doing it, but there may also be a plugin available that turns it on for you.

Those are some ideas. Please let us know what Santa Fe and Albuquerque admins are seeing.

**Comment by A46webchair** [comment]
This is getting a little out of my experience Zone but I will try... Since I just migrated to a new web hosting company (today) would there really be any bugs in the PHP? or - are those things not related?

**Comment by A46webchair** [comment]
Also ran a security program with my new host (SiteGround) that checks for blacklisting and our domain came back clean. Does that matter?

**Comment by tech2serve** [comment]
> This is getting a little out of my experience Zone but I will try... Since I just migrated to a new web hosting company (today) would there really be any bugs in the PHP? or - are those things not related?

Could be because I am suspecting a conflict between one of your other plugins (probably your site since two of us can get the feed from Santa Fe).

I admit, short of some weird configuration in your website that simply got migrated to your new hosting provider, I'm stumped. 

If your new hosting provider let's you create a second staging area site, you might try that and only add TSML. Skip all other configurations/plugins. Then see if you can add Santa Fe as a feed.

And I do think you'd benefit from close coordination with Santa Fe and Albuquerque to see if they see any errors when you try to get the feed.

**Comment by A46webchair** [comment]
@tech2serve 
Any chance you'd be willing to log in and look around in my WordPress? :) I will reach out to ABQ and Santa Fe but we are all about the same technical level - each of us have a little more than the others on certain things...

**Comment by A46webchair** [comment]
@tech2serve Where would Santa Fe and ABQ see an error when I try to get the feed for each one? Will the error display in the TSML plug-in somewhere?

**Comment by A46webchair** [comment]
@tech2serve Would this plug-in show us what we nee to see?

https://wordpress.org/plugins/debug-bar/

**Comment by tech2serve** [comment]
It looks like it will help, but you'll still need to modify your 'wp-config.php' file according to their plugin page:

```define( 'WP_DEBUG', true );
define( 'SAVEQUERIES', true );
```
Does you hosting provider allow you to create a second staging/test site?

**Comment by tech2serve** [comment]
> @tech2serve
> Any chance you'd be willing to log in and look around in my WordPress? :) I will reach out to ABQ and Santa Fe but we are all about the same technical level - each of us have a little more than the others on certain things...

I may be willing to do a Zoom or Teamviewer session with you. Please send an email to devs@code4recovery.org.

**Comment by A46webchair** [comment]
@tech2serve  I have set up a staging site. I will email you

**Comment by tech2serve** [comment]
On the staging site, with no other plugins other than TSML installed, are you able to connect either of the feeds?

**Comment by A46webchair** [comment]
Yes!! So I deactivated all plug-ins except TSML and the feeds are now working! 

 

How do I figure out the problem plug-in?

 

 

Kerri 

Area 46 Webchair

**Comment by tech2serve** [comment]
Usually one simply starts adding them back in one at a time until the problem resumes. I know that isn't very elegant, but it works especially when one isn't seeing any errors.

**Comment by A46webchair** [comment]
Progress - on the staging site I started adding back in one at a time.  This one "broke" the feeds. https://wordpress.org/plugins/quiz-master-next/

I deleted (not just deactivated) this on the live site but the feed still wont work.

Suggestions??

**Comment by A46webchair** [comment]
also... I have multiple data feeds listed on the live site (from several attempts) but the delete function is not working.. Plus the Data sources section looks different.  On the live site is compacted and on the staging site is stretches across the width of the page. Not sure if that matters...

**Comment by tech2serve** [comment]
I would still be looking for errors and conflicts. Since you using Firefox, press Option-Cmd-K to open the developer console. Your screen should look something like this:

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/31904316/97636615-60705580-1a07-11eb-8a6c-45dddaedc0ff.png">

Then navigate to your Data Sources and try deleting one of the unwanted feeds. Hopefully you'll get an error or some message that will give a clue as to why the process is failing.

**Comment by A46webchair** [comment]
@tech2serve   (I really appreciate all your help with this)
A little confused on exactly what you mean. The live site (admin) will not allow me to delete a data source. It is "stuck" for lack of a better way for me to explain it. Are either of these helpful? 

![image](https://user-images.githubusercontent.com/72042564/97637596-96f6a200-1a00-11eb-8192-659a6dce0abf.png)

![image](https://user-images.githubusercontent.com/72042564/97637764-dde49780-1a00-11eb-9906-8563d8af902a.png)

**Comment by A46webchair** [comment]
Woohoo!!!!!!! Meetings are restored.  This plug-in was the other culprit.  Thankfully I was not really using either one of them!!

https://wordpress.org/plugins/wordpress-simple-paypal-shopping-cart/

**Comment by tech2serve** [comment]
Here is what I understood: That when you click on the little "x` next to the timestamp of the last update of each feed, TSML did not delete that feed from the list. Is that correct?

Assuming yes, I can tell you the delete process works fine on my dev site, so I suspect the issue with this is related to your other feed troubles, and that it stems from one or much conflicts with other plugins. I do note from your screen shot above that a number of other plugins remain installed and probably active. I would suspect one of those.

I'll send you an email proposing a time to connect via Zoom. In the meantime, I would continue to remove/deactivate other plugins as I suspect when the real culprit is identified, it will become more apparent. And it is possible there is more than one conflict. I'm not crazy about seeing jQuery warnings in your console.

**Comment by A46webchair** [comment]
@tech2serve Thank you for your patience with me!
